packages/react/src/components/TimePicker/TimePicker.tsx

```diff
@@ -0,0 +1,193 @@
+      const optionText = onFormatDate ? onFormatDate(option) : formatTimeString(option, showSeconds, useHour12);
+      if (index === 0) {
+        setSelectedKey(optionText);
+      }
+      return {
+        key: optionText,
+        text: optionText,
+      };
+    });
+  }, [timeRange, increments, optionsCount, showSeconds, onFormatDate, useHour12]);
+
+  const onInputChange = React.useCallback(
+    (event: React.FormEvent<IComboBox>, option?: IComboBoxOption, index?: number, value?: string): void => {
+      if (onChange) {
+        onChange(event, option, index, value);
+      }
+
+      const validateUserInput = (userInput: string): string => {
+        let errorMessageToDisplay = '';
+        let regex: RegExp;
+        if (useHour12) {
+          regex = showSeconds ? REGEX_SHOW_SECONDS_HOUR_12 : REGEX_HIDE_SECONDS_HOUR_12;
+        } else {
+          regex = showSeconds ? REGEX_SHOW_SECONDS_HOUR_24 : REGEX_HIDE_SECONDS_HOUR_24;
+        }
+        if (!regex.test(userInput)) {
+          errorMessageToDisplay = strings.invalidInputErrorMessage;
+        }
+        return errorMessageToDisplay;
+      };
+
+      const key = option?.key;
+      let updatedUserText = '';
+      let errorMessageToDisplay = '';
+      if (value) {
+        if (allowFreeform && !option) {
+          if (!onFormatDate) {
+            // Validate only if user did not add onFormatDate
+            errorMessageToDisplay = validateUserInput(value);
+          } else {
+            // Use user provided validation if onFormatDate is provided
+            if (onValidateUserInput) {
+              errorMessageToDisplay = onValidateUserInput(value);
+            }
+          }
+        }
+        updatedUserText = value;
+      } else if (option) {
+        updatedUserText = option.text;
+      }
+
+      setErrorMessage(errorMessageToDisplay);
+      setUserText(updatedUserText);
+      setSelectedKey(key);
+    },
+    [
+      allowFreeform,
+      onFormatDate,
+      onValidateUserInput,
+      showSeconds,
+      useHour12,
+      onChange,
+      strings.invalidInputErrorMessage,
+    ],
+  );
+
+  const evaluatePressedKey = (event: React.KeyboardEvent<IComboBox>) => {
+    if (
+      !onFormatDate &&
+      // Only permit input of digits, space, colon, A/P/M characters
+      !(
+        (event.charCode >= KeyCodes.zero && event.charCode <= KeyCodes.colon) ||
+        event.charCode === KeyCodes.space ||
+        event.charCode === KeyCodes.a ||
+        event.charCode === KeyCodes.m ||
+        event.charCode === KeyCodes.p
+      )
+    ) {
+      event.preventDefault();
+    }
+  };
+
+  return (
+    <div>
+      <ComboBox
+        allowFreeform={allowFreeform}
+        selectedKey={selectedKey}
+        label={label}
+        errorMessage={errorMessage}
+        options={timePickerOptions}
+        onChange={onInputChange}
+        text={userText}
+        //eslint-disable-next-line
+        onKeyPress={evaluatePressedKey}
+        {...rest}
```

Comment:
It might be safer to spread these first, to ensure that your defined prop values take precedence

Reply:
Resolved by spreading `{...rest}` first.